### PR TITLE
Closes #1851: External Tools for ingested files fixes

### LINKED ISSUE #1851: Fixes for External Tools for ingested files
~~1. Open tool in new tab~~
2. Show «Explore» button only for public files
3. Do not show the button for tsv/csv files that were not ingested

### PR DESCRIPTION
**Pull request by wtr-icm**

### CODE REVIEW

File: dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/externaltools/ExternalToolServiceBean.java [hunk 1]
```diff
@@ -98,17 +107,39 @@ public ExternalTool save(ExternalTool externalTool) {
...
+
         allExternalTools.forEach((externalTool) -> {
-            //Match tool and file type 
+            // Match tool and file type
             if (contentType.equals(externalTool.getContentType())) {
+
+                // Exclude non-public tsv files and those which do not have datatable
+                if (isTsvAltContentType && (!isFilePublic(file, datasetVersion) || !file.isTabularData())) {
+                    return;
+                }
```
Comment (dbojanek) [inline dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/externaltools/ExternalToolServiceBean.java]: Can we please avoid using comments and use private methods with descriptive names instead?
Here for example something like `if (isTsvNonPublicOrNonTabular) ` or something like that.

File: dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/externaltools/ExternalToolServiceBean.java [hunk 2]
```diff
@@ -156,6 +187,16 @@ public static ExternalTool parseAddExternalToolManifest(String manifest) {
         return new ExternalTool(displayName, description, type, toolUrl, toolParameters, contentType);
     }
 
+    // -------------------- PRIVATE --------------------
+
+    private boolean isFilePublic(DataFile file, DatasetVersion datasetVersion) {
```
Comment (dbojanek) [inline dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/externaltools/ExternalToolServiceBean.java:191]: I believe this is intentional but just want to be sure: we only check if given file is public in specific dataset version and not across all versions? 
The consensus on a meeting was that we are ok with this being inconsistent with files endpoint?

File: dataverse-webapp/src/test/java/edu/harvard/iq/dataverse/externaltools/ExternalToolServiceBeanTest.java [hunk 3]
```diff
@@ -1,224 +1,262 @@
...
-        dataFile.setFileMetadatas(fmdl);
+        DatasetVersion datasetVersion = new DatasetVersion();
+        datasetVersion.setVersionState(released ? DatasetVersion.VersionState.RELEASED : DatasetVersion.VersionState.DRAFT);
+        Dataset dataset = new Dataset();
+        datasetVersion.setDataset(dataset);
+
+        Calendar calendar = Calendar.getInstance();
+        calendar.setTime(new Date());
+        calendar.add(Calendar.DAY_OF_MONTH, 1);
+        dataset.setEmbargoDate(embargoed ? calendar.getTime() : null);
```
Comment (dbojanek) [inline dataverse-webapp/src/test/java/edu/harvard/iq/dataverse/externaltools/ExternalToolServiceBeanTest.java]: Wouldn't it be more descriptive if we at least called calendar variable more like `futureEmbargoExpirationDate` or something?

File: dataverse-webapp/src/test/java/edu/harvard/iq/dataverse/externaltools/ExternalToolServiceBeanTest.java [hunk 4]
```diff
@@ -1,224 +1,262 @@
...
-        try {
-            ExternalTool externalTool = ExternalToolServiceBean.parseAddExternalToolManifest(null);
-        } catch (Exception ex) {
-            expectedException = ex;
-        }
-        assertNotNull(expectedException);
-        assertEquals("External tool manifest was null or empty!", expectedException.getMessage());
+    void parseAddExternalToolManifest__null() {
+        // when & then
+        assertThatThrownBy(() -> ExternalToolServiceBean.parseAddExternalToolManifest(null))
```
Comment (dbojanek) [inline dataverse-webapp/src/test/java/edu/harvard/iq/dataverse/externaltools/ExternalToolServiceBeanTest.java:172]: Can't we add here `.isInstanceOf()` ? What exception is expected?

File: dataverse-webapp/src/test/java/edu/harvard/iq/dataverse/externaltools/ExternalToolServiceBeanTest.java [hunk 5]
```diff
@@ -1,224 +1,262 @@
...
-        try {
-            ExternalTool externalTool = ExternalToolServiceBean.parseAddExternalToolManifest("");
-        } catch (Exception ex) {
-            expectedException = ex;
-        }
-        assertNotNull(expectedException);
-        assertEquals("External tool manifest was null or empty!", expectedException.getMessage());
+    void parseAddExternalToolManifest__emptyString() {
+        // when & then
+        assertThatThrownBy(() -> ExternalToolServiceBean.parseAddExternalToolManifest(""))
```
Comment (dbojanek) [inline dataverse-webapp/src/test/java/edu/harvard/iq/dataverse/externaltools/ExternalToolServiceBeanTest.java:180]: Same as above.

File: dataverse-webapp/src/test/java/edu/harvard/iq/dataverse/externaltools/ExternalToolServiceBeanTest.java [hunk 6]
```diff
@@ -1,224 +1,262 @@
...
-        try {
-            ExternalTool externalTool = ExternalToolServiceBean.parseAddExternalToolManifest(tool);
-        } catch (Exception ex) {
-            expectedException = ex;
-        }
-        assertNotNull(expectedException);
-        assertEquals("Unknown reserved word: mode1", expectedException.getMessage());
+
+        // when & then
+        assertThatThrownBy(() -> ExternalToolServiceBean.parseAddExternalToolManifest(tool))
```
Comment (dbojanek) [inline dataverse-webapp/src/test/java/edu/harvard/iq/dataverse/externaltools/ExternalToolServiceBeanTest.java:210]: Here as well.

Comment (dbojanek) [reply dataverse-webapp/src/test/java/edu/harvard/iq/dataverse/externaltools/ExternalToolServiceBeanTest.java:210]: And below where we test exception being thrown.

File: dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/externaltools/ExternalToolServiceBean.java [hunk 7]
```diff
@@ -98,17 +107,39 @@ public ExternalTool save(ExternalTool externalTool) {
         return em.merge(externalTool);
     }
 
+    // Used only in REST
```
Comment (dbojanek) [inline dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/externaltools/ExternalToolServiceBean.java]: IMO this comment is not helpful. We can check where it is being used in IDE, but it'll be hard to maintain if we use this method elsewhere. If it's meant to be used only in REST maybe change the comment to `Should be only used in REST`, if not just remove it pls.

File: dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/externaltools/ExternalToolServiceBean.java (hunk 2 above)
Comment (wtr-icm) [reply dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/externaltools/ExternalToolServiceBean.java:191]: As far as I remember we decided to check only in context of the version the user is viewing.

**Review by dbojanek — APPROVED** [review]

**Pull request merged**